compiler/src/dotty/tools/dotc/ast/TreeInfo.scala

```diff
@@ -510,7 +510,7 @@ trait TypedTreeInfo extends TreeInfo[Type] { self: Trees.Instance[Type] =>
    */
   def constToLiteral(tree: Tree)(implicit ctx: Context): Tree = {
     val tree1 = ConstFold(tree)
-    tree1.tpe.widenTermRefExpr.dealias match {
+    tree1.tpe.widenTermRefExpr.dealias.normalized match {
```

Comment:
```suggestion
    tree1.tpe.widenTermRefExpr.normalized match {
```

I'm wondering if it will make a difference.

Reply:
It does not work without the dealiasing

Reply:
Maybe we should refine `normalized`.

Reply:
@OlivierBlanvillain should `normalized` dealias?